controller/src/main/java/org/jboss/as/controller/CapabilityReferenceRecorder.java

```diff
@@ -76,6 +77,23 @@ default boolean isDynamicDependent() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Returns the elements to be added to the baseRequirementName to build the capability name pattern.
+     * The pattern is of the form `$segment[.$segment]` where each segment represents either the name of one of the
```

Comment:
Not: The form is now `segment[.segment]`. The RRD output is now a LIST of STRING with the $ char no longer relevant.